source/agora/consensus/data/Transaction.d

```diff
@@ -247,3 +248,44 @@ public bool getSumOutput (const Transaction tx, ref Amount acc)
             return false;
     return true;
 }
+
+/*******************************************************************************
+
+    Caculates the fee of transaction data to store
```

Comment:
I think this function should not calculate the fees, but only a **multiplier**. 

Think about the following: if BOA is worth 1$ today, but 100$ next year, do you really want the fee to be hardcoded into the protocol?

Fees may be dynamic and also entirely up to the validator nodes. But it's good to have a helper function like this to help most validators pick the fees.

Reply:
There was something like this on the issue.
```
### High level requirements:

- It must come at a cost, which might not be linear (to allow small packet of data but forbid images & co, at least in a contiguous chunk);

### Suggested approach

5. If those rules are satisfied, the transaction can contain a certain amount of data, of which the size is dependent on the amount in the output towards the common budget;
```

If this value is determined by the validator's node, how can transaction validation be done?
For your information, I have already implemented code in PR #1258 that validates according to the amount of transaction data. Please check.

Reply:
I'm trying to save the factor in consensusParams to compensate for the variation in the value of BOA. How is it?

Reply:
Alright, I misunderstood what was written.

I'm still not sure how this can work by being hardcoded. But you don't have to change your code.